src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_state_manager.ts

```diff
@@ -336,7 +336,12 @@ export class DashboardStateManager {
     if (isLegacyStringQuery) {
       return lastSavedQuery !== query.query;
     }
-
+    if (lastSavedQuery && typeof currentQuery !== 'string') {
+      const currQuery = currentQuery as Query;
+      if (currQuery.query === lastSavedQuery.query) {
```

Comment:
If we only change the language on the empty query, before applying updates, no need to mark the dashboard as dirty (I think).